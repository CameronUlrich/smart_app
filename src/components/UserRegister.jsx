import React, { Component } from 'react';
import { withRouter} from 'react-router-dom'
import { render } from 'react-dom';  // need to install react-dom on server
import { Helmet } from 'react-helmet';
import { ToastContainer, toast } from 'react-toastify';
import cookie from 'react-cookies';

import './styles/Buttons.css';
import './styles/Titles.css';

import './styles/UserRegister.css';

class UserRegister extends Component {

    constructor(props) {
        super(props);
        this.state = {
            username: '',
            password: '',
            email: '',
            exists: false
        };
    }


    // updates the email with whatever is in the email field
    updateEmail(e) {
        this.setState({
            email: e.target.value
        })
    }

    // updates the username with whatever is in the username field
    updateUsername(e) {
        this.setState({
            username: e.target.value
        })
    }

    // updates the password with whatever is in the password field
    updatePassword(e) {
        this.setState({
            password: e.target.value
        })
    }

    registerUser = () => {
        this.props.history.push('/');
    }

    handleClick(e) {
        e.preventDefault();
        console.log('The button was clicked.');
        this.userRegister(this.state.email, this.state.username, this.state.password);
    }

    userRegister(email, username, password) {
        fetch('http://localhost:3001')
        .then(response => {
            return response.json();
        })
        .then(data => {
            data.forEach(row => {
                // checks to see if the email exists in the database
                if (row.email === email) {
                    alert("Email is already registered under another account. Please choose another.");
                    this.state.exists = true;
                }
                // checks to see if the username exists in the database
                else if (row.username === username){
                    alert("Username is already registered under another account. Please choose another.");
                    this.state.exists = true;
                }
                else {
                    this.state.exists = false;
                }
            })
        })
        if (!this.state.exists){
            fetch('http://localhost:3001/createUser', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                },
                body: JSON.stringify({email, username, password}),
            })
            .then(response => {
                return response.json();
            })
            .then(data => {
                console.log(data);
                this.registerUser();
            });
        }
    }
    

    render() {
        

        return (
            <div>
                <ToastContainer style={{fontSize: "20px", width: "450px", top: "100px" }} />

                <Helmet>
                    <style>{'body { background-color: #282c34; }'}</style>
                </Helmet>

                <br></br>
                <br></br>
                <br></br>

                <h1 id="registerText" className="welcome">Register</h1>

                <text id="userText">Email: </text>
                

                <input id="email" type="text" placeholder="Email" value={this.state.email} onChange={e => this.updateEmail(e)}></input>

                <br></br>
                <br></br>
                
                <text id="userText">Username: </text>
                

                <input id="userName" type="text" placeholder="Username" value={this.state.username} onChange={e => this.updateUsername(e)}></input>

                <br></br>
                <br></br>

                <text id="passText">Password:  </text>
                

                <input id="password" type="text" placeholder="Password" value={this.state.password} onChange={e => this.updatePassword(e)}></input>
                
                <br></br>
                <br></br>
                <br></br>

                <button id="createBtn" onClick={e => this.handleClick(e)}>Create account</button>


            </div>

        );

        
    
    }
}

export default withRouter(UserRegister);