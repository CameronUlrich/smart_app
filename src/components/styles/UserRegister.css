#registerText{
    animation: fadeIn 1s ease-out;
}

#email {

    position: relative;
    left: auto;
    top: auto;

    width: 250px;
    box-sizing: border-box;
    border: 2px solid #ccc;
    border-radius: 4px;
    font-size: 20px;
    background-color: transparent;

    color: white;
    outline: none;
    
    background-position: 10px 10px; 
    background-repeat: no-repeat;
    padding: 12px 20px 12px 10px;
    -webkit-transition: width 0.4s ease-in-out;
    transition: width 0.4s ease-in-out;
    animation: fadeIn 1s ease-out;
}

#userName {

    position: relative;
    left: auto;
    top: auto;

    width: 250px;
    box-sizing: border-box;
    border: 2px solid #ccc;
    border-radius: 4px;
    font-size: 20px;
    background-color: transparent;

    color: white;
    outline: none;
    
    background-position: 10px 10px; 
    background-repeat: no-repeat;
    padding: 12px 20px 12px 10px;
    -webkit-transition: width 0.4s ease-in-out;
    transition: width 0.4s ease-in-out;
    animation: fadeIn 1s ease-out;
}

#password {

    position: relative;
    left: auto;
    top: auto;

    width: 250px;
    box-sizing: border-box;
    border: 2px solid #ccc;
    border-radius: 4px;
    font-size: 20px;
    background-color: transparent;

    color: white;
    outline: none;
    
    background-position: 10px 10px; 
    background-repeat: no-repeat;
    padding: 12px 20px 12px 10px;
    -webkit-transition: width 0.4s ease-in-out;
    transition: width 0.4s ease-in-out;
    animation: fadeIn 1s ease-out;
}

#userText{
    position: relative;
    left: -5px;
    font-size: 20px;
    color: white;
    animation: fadeIn 1s ease-out;
}

#passText{
    position: relative;
    left: -5px;
    font-size: 20px;
    color: white;
    animation: fadeIn 1s ease-out;
}




#createBtn{
    position: relative;
    padding: 10px 26px;
    top: auto;
    border: none;
    background-color: transparent;
    border: 2px solid rgb(255, 255, 255);
    left: auto;
    color: white;
    
    
    font-size: 19px;
    text-decoration: none;
    background-color: transparent;
    -webkit-transition-duration: 0.4s; /* Safari */
    transition-duration: 0.4s;
    cursor: pointer;
    animation: fadeIn 1s ease-out;
    

}

#createBtn:hover{
    border: none;
    background-color: rgba(13, 78, 255, 0.171);
    
    border: 2px solid white;
    border-radius: 25px;
    color: white;
    
    font-size: 19px;
    text-decoration: none;
}

@keyframes fadeIn{
    0% {
        opacity: 0;
        transform: translateY(5%);
    } 
    100% {
        opacity: 1;
        transform: translateY(0%);
    }
}