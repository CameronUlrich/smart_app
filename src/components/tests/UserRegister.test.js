import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import UserRegister from '../UserRegister';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() })

it("renders without crashing", ()=>{
    const div = document.createElement("div");
    ReactDOM.render(<BrowserRouter><UserRegister></UserRegister></BrowserRouter>, div)
})



test('render a small label', () => {
    const wrapper = shallow(
        <UserRegister></UserRegister>
    );
    expect(wrapper).toMatchSnapshot();
});
