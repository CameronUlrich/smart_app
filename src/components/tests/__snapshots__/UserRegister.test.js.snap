// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRegister render a small label 1`] = `ShallowWrapper {}`;

exports[`UserRegister should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`render a small label 1`] = `ShallowWrapper {}`;
